go/vt/vtctl/workflow/utils.go

```diff
@@ -1011,3 +1011,73 @@ func applyTargetShards(ts *trafficSwitcher, targetShards []string) error {
 	}
 	return nil
 }
+
+// validateEmptyTables checks if all specified tables in the keyspace are empty across all shards.
+// It queries each shard's primary tablet and if any non-empty table is found, it returns an error
+// containing a list of non-empty tables.
+func validateEmptyTables(ctx context.Context, ts *topo.Server, tmc tmclient.TabletManagerClient, keyspace string, tables []string) error {
+	if len(tables) == 0 {
+		return nil
+	}
+
+	shards, err := ts.GetServingShards(ctx, keyspace)
+	if err != nil {
+		return err
+	}
+	if len(shards) == 0 {
+		return fmt.Errorf("keyspace %s has no shards", keyspace)
+	}
+
+	var selectQueries []string
+	for _, t := range tables {
+		selectQueries = append(selectQueries, fmt.Sprintf("(select '%s' from %s limit 1)", t, t))
+	}
+	query := strings.Join(selectQueries, "union all")
```

Comment:
I think we need a way to handle cases where there are many tables as the number of tables is unbounded. Two factors we'll have to consider:
 1. This one statement could be so long that it is beyond `max_allowed_packet`
 2. This one statement could affect more than the maximum allowed tables 

In either case, the user is stuck and there is no outlet. At least unless we add a new flag that skips this test / allows for tables with existing data.